runtime/doc/ui.txt

```diff
@@ -249,6 +248,14 @@ the editor.
 	Indicates to the UI that it must stop rendering the cursor. This event
 	is misnamed and does not actually have anything to do with busyness.
 
+["restart", progpath, argv] ~
+	|:restart| command has been used and the Nvim server is about to exit.
+	The UI should start a new Nvim server using `progpath` as the full path
+	to the Nvim executable |v:progpath| and `argv` as its command-line
+	arguments |v:argv|, and then reattach to the new server.
+	Note: |--embed| and |--headless| are excluded from `argv`, and the client
```

Comment:
I'm not entirely sure if this should be done on the client or the server, but considering the future addition of `:restart [args]`, I'm inclined to do this on the server, since custom `[args]` usually don't include `--embed` or `--headless` either.

Reply:
Strongly prefer doing as much as possible on the server, in general.